### PR TITLE
Add name field to a rule

### PR DESCRIPTION
**Pull request by kyessenov**
This is necessary to make a route rule a self-contained object.

### CODE REVIEW

File: proxy/v1/config/route_rule.proto [hunk 1]
```diff
@@ -72,6 +72,10 @@ package istio.proxy.v1.config;
 //       weight: 100
 //
 message RouteRule {
+  // REQUIRED: Route rules have unique names to allow multiple rules
+  // for the same destination, e.g. "my-rule".
+  string name = 11;
```
Comment (mandarjog) [inline proxy/v1/config/route_rule.proto]: You were against encoding the name inside the proto.
now it is double encoded `/route-rule/{destination}/{rule-name}`

Comment (kyessenov) [inline proxy/v1/config/route_rule.proto]: The machinery to embed metadata fields into protos is difficult. There are no traits in proto lang or intefaces AFAIK. Also, not all config types have names.

**Comment by kyessenov** [comment]
PTAL

**Comment by kyessenov** [comment]
Destination policy does not need a name, so no changes there.

**Pull request merged**